weldx/tests/test_time.py

```diff
@@ -0,0 +1,127 @@
+"""Test the `Time` class."""
+
+from typing import List
+
+import numpy as np
+import pint
+import pytest
+from pandas import DatetimeIndex, Timedelta, TimedeltaIndex, Timestamp
+
+from weldx import Q_
+from weldx.time import Time
+
+
+def _initialize_type(cls_type, values, unit):
+    """Initialize the passed time type."""
+    if cls_type not in (pint.Quantity, Timedelta) and not isinstance(values, List):
+        values = [values]
+    if cls_type is np.timedelta64:
+        return np.array(values, dtype=f"timedelta64[{unit}]")
+    if cls_type is Time:
+        return Time(Q_(values, unit))
+    return cls_type(values, unit)
+
+
+class TestTime:
+    """Test the time class."""
+
+    # test_init ------------------------------------------------------------------------
+
+    @staticmethod
+    @pytest.mark.parametrize(
+        "time, time_ref, exp_absolute, exp_time_ref",
+        [
+            (TimedeltaIndex([1, 2, 3]), None, False, None),
+            (TimedeltaIndex([1, 2, 3]), Timestamp("16:00"), True, "16:00"),
+            (TimedeltaIndex([1, 2, 3]), "16:00", True, "16:00"),
+            (Q_([1, 2, 3], "s"), None, False, None),
+            (Q_(2, "s"), None, False, None),
+            (Q_([1, 2, 3], "s"), "16:00", True, "16:00"),
+            (np.array([1, 2, 3], dtype="timedelta64[s]"), None, False, None),
+            (np.array([1, 2, 3], dtype="timedelta64[s]"), "16:00", True, "16:00"),
+            (DatetimeIndex(["2000", "2010"]), None, True, "2000"),
+            (DatetimeIndex(["2000", "2010"]), "2005", True, "2005"),
+            (np.array(["2000", "2010"], dtype="datetime64"), None, True, "2000"),
+            (np.array(["2000", "2010"], dtype="datetime64"), "2005", True, "2005"),
+            (Timestamp("2000"), None, True, "2000"),
+            (Timestamp("2000"), "2005", True, "2000"),
+            (Timedelta(1, "d"), None, False, None),
+            (Timedelta(1, "d"), "2005", True, "2005-01-02"),
+            (np.datetime64("2000"), None, True, "2000"),
+            (np.datetime64("2000"), "2005", True, "2000"),
+            (np.timedelta64(1, "s"), None, False, None),
+            (np.timedelta64(1, "s"), "16:00", True, "16:00:01"),
```

Comment:
We should rediscuss the behavior in this case here. It differs from the array behavior. We put in a time delta with a reference time, which gets turned into a Timestamp. Now we ignore the reference time. If we would call `as_timedelta`, it would return `"0s"`. Get's even worse if we have a single value array like `np.timedelta64([1], "s")` which will also be converted to a timestamp. On the other hand, `np.timedelta64([1, 2], "s")` will return `"1s"` and `"2s"` . This inconsistency can lead to serious hard to find bugs in some kind of automatic processing. So it boils down to also storing and handling reference times for timestamps.

Reply:
reference time should also be respected for timestamps, true 👍 
should be fixed now

Reply:
Had to remove one more line in the `__init__`.